modules/dnn/src/layers/cpu_kernels/softmax_kernels.default.hpp

```diff
@@ -0,0 +1,166 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+// This file is modified from the ficus (https://github.com/vpisarev/ficus/blob/master/lib/NN/OpNN.fx).
+// Here is the original license:
+/*
+    This file is a part of ficus language project.
+    See ficus/LICENSE for the licensing terms
+*/
+
+#include <opencv2/core/hal/intrin.hpp>
+#include <opencv2/core/utility.hpp> // parallel_for_
+
+#if CV_SIMD128
+#define _c_exp_hi 88.3762626647949f
+#define _c_exp_lo -88.3762626647949f
+
+#define _c_cephes_LOG2EF 1.44269504088896341
+#define _c_cephes_exp_C1 0.693359375
+#define _c_cephes_exp_C2 -2.12194440e-4
+
+#define _c_cephes_exp_p0 1.9875691500E-4
+#define _c_cephes_exp_p1 1.3981999507E-3
+#define _c_cephes_exp_p2 8.3334519073E-3
+#define _c_cephes_exp_p3 4.1665795894E-2
+#define _c_cephes_exp_p4 1.6666665459E-1
+#define _c_cephes_exp_p5 5.0000001201E-1
+
+#define _VEXP_INIT_DEFAULT() \
+    v_float32x4 _vexp_lo = v_setall_f32(_c_exp_lo); \
+    v_float32x4 _vexp_hi = v_setall_f32(_c_exp_hi); \
+    v_float32x4 _vexp_half = v_setall_f32(0.5f); \
+    v_float32x4 _vexp_one = v_setall_f32(1.f); \
+    v_float32x4 _vexp_LOG2EF = v_setall_f32(_c_cephes_LOG2EF); \
+    v_float32x4 _vexp_C1 = v_setall_f32(-_c_cephes_exp_C1); \
+    v_float32x4 _vexp_C2 = v_setall_f32(-_c_cephes_exp_C2); \
+    v_float32x4 _vexp_p0 = v_setall_f32(_c_cephes_exp_p0); \
+    v_float32x4 _vexp_p1 = v_setall_f32(_c_cephes_exp_p1); \
+    v_float32x4 _vexp_p2 = v_setall_f32(_c_cephes_exp_p2); \
+    v_float32x4 _vexp_p3 = v_setall_f32(_c_cephes_exp_p3); \
+    v_float32x4 _vexp_p4 = v_setall_f32(_c_cephes_exp_p4); \
+    v_float32x4 _vexp_p5 = v_setall_f32(_c_cephes_exp_p5)
+
+#define _VEXP_COMPUTE_DEFAULT(x, y) { \
+    v_float32x4 _vexp_, _vexp_x, _vexp_y, _vexp_z; \
+    _vexp_x = v_min(x, _vexp_hi); \
+    _vexp_x = v_max(_vexp_x, _vexp_lo); \
+    _vexp_ = v_fma(_vexp_x, _vexp_LOG2EF, _vexp_half); \
+    v_int32x4 _vexp_mm = v_floor(_vexp_); \
+    _vexp_ = v_cvt_f32(_vexp_mm); \
+    _vexp_mm = v_add(_vexp_mm, v_setall_s32(0x7f)); \
+    _vexp_mm = v_shl(_vexp_mm, 23); \
+    _vexp_x = v_fma(_vexp_, _vexp_C1, _vexp_x); \
+    _vexp_x = v_fma(_vexp_, _vexp_C2, _vexp_x); \
+    _vexp_z = v_mul(_vexp_x, _vexp_x); \
+    _vexp_y = v_fma(_vexp_x, _vexp_p0, _vexp_p1); \
+    _vexp_y = v_fma(_vexp_y, _vexp_x, _vexp_p2); \
+    _vexp_y = v_fma(_vexp_y, _vexp_x, _vexp_p3); \
+    _vexp_y = v_fma(_vexp_y, _vexp_x, _vexp_p4); \
+    _vexp_y = v_fma(_vexp_y, _vexp_x, _vexp_p5); \
+    _vexp_y = v_fma(_vexp_y, _vexp_z, _vexp_x); \
+    _vexp_y = v_add(_vexp_y, _vexp_one); \
+    y = v_mul(_vexp_y, v_reinterpret_as_f32(_vexp_mm)); \
+    }
+#endif
+
+namespace cv { namespace dnn { namespace cpu_baseline {
+
+void softmax(const Mat &src, Mat &dst, int axis);
```

Comment:
`softmax` -> `softmaxKernel` to avoid confusion.

Reply:
👌